application-ckeditor-plugins/src/main/resources/xwiki-image/plugin.js

```diff
@@ -31,8 +31,12 @@
     beforeInit: function(editor) {
       editor.on('widgetDefinition', function (event) {
         var widgetDefinition = event.data;
-        if (widgetDefinition.name === "image" && widgetDefinition.dialog === "image2") {
-          this.overrideImageWidget(editor, widgetDefinition);
+        if (widgetDefinition.name === "image") {
+          // Adds support for configuring the allowed content for image caption (currently hard-coded).
+          event.data.editables.caption.allowedContent = event.editor.config['xwiki-image'].captionAllowedContent;
```

Comment:
```suggestion
          widgetDefinition.editables.caption.allowedContent = (editor.config['xwiki-image'] || {}).captionAllowedContent;
```

I think ``editor.config['xwiki-image']`` can be null if you don't specify any configuration.

Also, is there any reason for not putting this inside ``overrideImageWidget``?